systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessST.java

```diff
@@ -183,6 +284,70 @@ void testMirrorMaker2RackAwareness(ExtensionContext extensionContext) {
         assertThat(commandOut.equals("consumer.client.rack=" + hostname), is(true));
+
+        // Mirroring messages by: Producing to the Source Kafka Cluster and consuming them from mirrored KafkaTopic in target Kafka Cluster.
+
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(kafkaClusterSourceName, testStorage.getTopicName(), 3).build());
+
+        LOGGER.info("Producing messages into the source Kafka: {}/{}, topic: {}", testStorage.getNamespaceName(), kafkaClusterSourceName, testStorage.getTopicName());
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(kafkaClusterSourceName))
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .build();
+        resourceManager.createResource(extensionContext, clients.producerStrimzi());
+        ClientUtils.waitForProducerClientSuccess(testStorage);
+
+        LOGGER.info("Consuming messages in the target Kafka: {}/{} mirrored topic: {}", testStorage.getNamespaceName(), kafkaClusterTargetName, sourceMirroredTopicName);
+        clients = new KafkaClientsBuilder(clients)
+            .withTopicName(sourceMirroredTopicName)
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(kafkaClusterTargetName))
+            .build();
+        resourceManager.createResource(extensionContext, clients.consumerStrimzi());
+        ClientUtils.waitForConsumerClientSuccess(testStorage);
+    }
+
+    private void produceConsumeDataWithSinkConnector(ExtensionContext extensionContext, TestStorage testStorage) {
+
+        // Deploy Kafka Connector
+        Map<String, Object> connectorConfig = new HashMap<>();
+        connectorConfig.put("topics", testStorage.getTopicName());
+        connectorConfig.put("file", Constants.DEFAULT_SINK_FILE_PATH);
+        connectorConfig.put("key.converter", "org.apache.kafka.connect.storage.StringConverter");
+        connectorConfig.put("value.converter", "org.apache.kafka.connect.storage.StringConverter");
+
+        LOGGER.info("Creating KafkaConnector: {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        resourceManager.createResource(extensionContext, KafkaConnectorTemplates.kafkaConnector(testStorage.getClusterName())
+            .editMetadata()
+                .withNamespace(testStorage.getNamespaceName())
+            .endMetadata()
+            .editSpec()
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .withConfig(connectorConfig)
+            .endSpec()
+            .build());
+        KafkaConnectorUtils.waitForConnectorReady(testStorage.getNamespaceName(), testStorage.getClusterName());
```

Comment:
You don't need to wait for KafkaConnector readiness, as it is already done in the `createResource` method

Reply:
I agree, I think that is part of this issue - https://github.com/strimzi/strimzi-kafka-operator/issues/7368 .

I would pick `createResourceWithWait` and `createResourceWithoutWait`.

Reply:
if all these usages are only about waiting for Ready state,  `createResourceWithWait`, `createResourceWithoutWait`  sounds reasonable to me.

Reply:
Okay, I have done a job for ResourceManager here -> https://github.com/strimzi/strimzi-kafka-operator/pull/8634